swap/src/protocol/alice/message5.rs

```diff
@@ -73,12 +74,11 @@ impl NetworkBehaviourEventProcess<RequestResponseEvent<BobToAlice, AliceToBob>>
                     },
                 ..
             } => {
-                if let BobToAlice::Message5(msg) = request {
-                    debug!("Received message 5");
-                    self.events.push_back(OutEvent::Msg(msg));
-                    // Send back empty response so that the request/response protocol completes.
-                    self.rr.send_response(channel, AliceToBob::Message5);
-                }
+                let Request::Message5(msg) = request;
+                debug!("Received message 5");
+                self.events.push_back(OutEvent::Msg(msg));
+                // Send back empty response so that the request/response protocol completes.
+                let _ = self.rr.send_response(channel, Response::Message5);
```

Comment:
I think you shouldn't need to send back an empty response, you can just close the substream after writing / reading to it (should probably happen in the codec).